target/linux/ath79/patches-4.19/411-mtd-cfi_cmdset_0002-force-word-write.patch

```diff
@@ -0,0 +1,61 @@
+--- a/drivers/mtd/chips/cfi_cmdset_0002.c
++++ b/drivers/mtd/chips/cfi_cmdset_0002.c
+@@ -40,7 +40,7 @@
+ #include <linux/mtd/xip.h>
+ 
+ #define AMD_BOOTLOC_BUG
+-#define FORCE_WORD_WRITE 0
++#define FORCE_WORD_WRITE 1
```

Comment:
Please also provide a patch for 5.4. Despite, this looks like it may have side effects on other devices?

Reply:
Will do. At the moment it looks like this device is the only one using cfi-flash under ath79.

Reply:
Please add this as separate commit with proper commit description, why is this change needed.  Shouldn't you add your device into `cfi_fixup_table`? @rmilecki can you please check this?

Reply:
Could be. I'll have to figure how to use it first.

Reply:
This is hacky and may be harmful for other devices. Please just add a proper entry to the `cfi_fixup_table` as suggested by @ynezz. It should result in the same behaviour. All you need should be a line like:
```
{ CFI_MFR_XXX, 0x1234, fixup_use_write_buffers },
```

Reply:
Thank you for a concrete proposal. However, what it looks like I am trying to achieve the direct opposite. I want to use cfi_amdstd_write_words as write method and the code below makes all device use method cfi_amdstd_write_buffers. 
```
#if !FORCE_WORD_WRITE
	{ CFI_MFR_ANY, CFI_ID_ANY, fixup_use_write_buffers },
#endif
```
Turning on the flag does what I want. I tried to implement my own fixup_ method and inserted an entry in the cfi_fixup_table after the above entry but I did not work. 

Do you have any ideas how to turn on debug prints for this given file? If that's not possible, how turn them on generally? Another related question is that why a patch that turns the upstream flag on is suboptimal solution?